src/panels/lovelace/cards/hui-entity-button-card.ts

```diff
@@ -144,7 +147,12 @@ class HuiEntityButtonCard extends LitElement implements LovelaceCard {
         ${this._config.show_icon
           ? html`
               <ha-icon
-                data-domain=${computeStateDomain(stateObj)}
+                data-domain=${ifDefined(
+                  this._config.state_color ||
+                    (domain === "light" && this._config.state_color !== false)
```

Comment:
Why check for the light domain? It doesn't matter here, does it?

Reply:
no, it should not as the default is true. I'll change it